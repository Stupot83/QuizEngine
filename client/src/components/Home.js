import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import QuizInterface from "./QuizInterface";
import "../sass/Home.scss";

class Home extends Component {
    render() {
        return (
            <div className="Home">
                <QuizInterface />
            </div>
        );
    }
}

Home.propTypes = {
    auth: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
    auth: state.auth
});

export default connect(mapStateToProps)(withRouter(Home));
