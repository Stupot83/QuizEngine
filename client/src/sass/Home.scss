.Home {
    height: 100vh;
    background-image: url("https://www.transparenttextures.com/patterns/brick-wall.png");
}
