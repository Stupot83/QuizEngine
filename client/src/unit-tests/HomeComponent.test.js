import React from "react";
import { shallow } from "enzyme";
import Home from "../components/Home";
import QuizEngineStore from "../quizEngineStore";
import { Provider } from "react-redux";

const mockStore = QuizEngineStore;

describe("Home component", () => {
    it("should render correctly in debug mode", () => {
        const component = shallow(
            <Provider store={mockStore}>
                <Home debug />
            </Provider>
        );

        expect(component).toMatchSnapshot();
    });

    it("should render correctly", () => {
        const component = shallow(
            <Provider store={mockStore}>
                <Home />
            </Provider>
        );

        expect(component).toMatchSnapshot();
    });
});
