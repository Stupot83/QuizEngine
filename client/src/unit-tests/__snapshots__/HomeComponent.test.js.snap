// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render correctly 1`] = `ShallowWrapper {}`;

exports[`Home component should render correctly in debug mode 1`] = `ShallowWrapper {}`;
